### PR TITLE
feat(GuildManager): allow system channel flags in create

### PR DESCRIPTION
**Pull request by almostSouji**
**Please describe the changes this PR makes and why it should be merged:**

Allow `systemChannelFlags` to be passed during guild creation as per https://github.com/discord/discord-api-docs/pull/1764

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by vladfrangu — APPROVED** [review]

File: typings/index.d.ts [hunk 1]
```diff
@@ -2761,6 +2761,7 @@ declare module 'discord.js' {
     roles?: PartialRoleData[];
     systemChannelID?: number;
     verificationLevel?: VerificationLevel | number;
+    systemChannelFlags?: SystemChannelFlagsResolvable;
```
Comment (kyranet) [inline typings/index.d.ts:2764]: Please move this property above verificationLevel so they're sorted alphabetically.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by SpaceEEC** [comment]
I'm unable to get this working, the system channel flags are always 0 in the response body / guild create payload.

**Comment by almostSouji** [comment]
- the data we receive from `POST /guilds` has the correct flags set (3)
- the data we receive from the `GUILD_CREATE` websocket event has no flags set (0)
- getting data after the fact from `GET /guilds/:id` also has the correct flags set (3)
- the setting is applied, but the client seems to encounter that issue as well (the toggles show that the messages are enabled, they are however not sent)

So in conclusion that seems to be a discord bug and nothing we can handle on our end (if the GC arrives earlier we use that data in the #create response)

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]
We'll see if Discord can resolve this.
